src/textual/reactive.py

```diff
@@ -50,6 +53,108 @@ class TooManyComputesError(ReactiveError):
     """Raised when an attribute has public and private compute methods."""
 
 
+class WatchDecorator(Generic[WatchMethodType]):
+    """Watch decorator.
+
+    Decorate a method to make it a watcher.
+
+    """
+
+    def __init__(self, reactive: Reactive | None = None) -> None:
+        self._reactive = reactive
+
+    @overload
+    def __call__(self, *, init: bool = True) -> WatchDecorator[WatchMethodType]: ...
+
+    @overload
+    def __call__(self, method: WatchMethodType) -> WatchMethodType: ...
+
+    def __call__(
+        self, method: WatchMethodType | None = None, *, init: bool = True
+    ) -> WatchMethodType | WatchDecorator[WatchMethodType]:
+        _rich_traceback_omit = True
+        if method is None:
+            return self
```

Comment:
If we use the decorator as `@some_reactive.watch(init=False)` we need to save the value of `init=False` _somewhere_ before returning the decorator, otherwise `init` is always used as `True`.

To test this, try using `some_reactive.watch(init=False)` around a method and print the value of `init` after the assertions. The value `True` will be printed.

This means we also need a test to make sure that `init=False` is respected when specified in the watch decorator.

Reply:
Good catch. Turns out the init wasn't needed at all. I've removed it.

Reply:
Ok, good!